### DISCUSSION TITLE
Protected branch and commits

### DISCUSSION THREAD

**Discussion by jesperthoeger** [community/community#22566] [New to GitHub]
<p>Im using the GitHub extension for Visual Studio.</p>
<p>I have a project with a protected branch (master).</p>
<p>A friend of my have cloned the project through visual studio, and forgot to make a new branch before making a change and a commit.</p>
<p>He cannot push this commit to GitHub. Which is good.</p>
<p>But how can he remove the change/commit he made locally, so it will disapear from the commits to sync?</p>

**Comment by Wabri** [comment]
<p>Maybe you need the stash command: <a href="https://git-scm.com/book/en/v1/Git-Tools-Stashing" rel="nofollow noopener">git-stashing</a></p>

**Comment by mpboom** [comment]
<p>Nope, stashing is used to save current work temporarily (on a stash, hence the name).</p>
<p><strong>WARNING:</strong></p>
<p><strong>The command below will PERMANTENTLY delete the commit! Use with caution - never execute commands random people on the internet suggest without having a backup and knowing what they do!</strong></p>
<p>Try:</p>
<pre><code>git reset HEAD~1
</code></pre>
<p>This will undo the last commit and leave your current changes as they are.</p>

**Comment by Wabri** [comment, marked as answer]
<p>You are right… But instead of stashing he can create a new branch from the local master and push to origin this new branch, then reset the local master to origin/master.  Whit this method no modify were lost and they can review from a PR in github.</p>
<p>In line there are the commands you need to run:</p>
<pre><code>git checkout -B new_branch
</code></pre>
<p>this command will create a new branch keeping all the changes made.</p>
<p>Then push this new_branch to origin:</p>
<pre><code>git push origin new_branch
</code></pre>
<p>Return to master:</p>
<pre><code>git checkout master
</code></pre>
<p>and reset with the origin master with:</p>
<pre><code>git reset --hard HEAD^
</code></pre>
<p>This done the trick.</p>
<p>-Gabriele-</p>
